core/schema/secret.go

```diff
@@ -94,6 +101,52 @@ func (s *secretSchema) setSecret(ctx context.Context, parent *core.Query, args s
 	return i, nil
 }
 
+type mapSecretArgs struct {
+	Name string
+	URI  string
+}
+
+func (s *secretSchema) mapSecret(ctx context.Context, parent *core.Query, args mapSecretArgs) (i dagql.Instance[*core.Secret], err error) {
+	clientMetadata, err := engine.ClientMetadataFromContext(ctx)
+	if err != nil {
+		return i, fmt.Errorf("failed to get client metadata from context: %w", err)
+	}
+
+	secretStore, err := parent.Secrets(ctx)
+	if err != nil {
+		return i, fmt.Errorf("failed to get secret store: %w", err)
+	}
+
+	accessor, err := core.GetClientResourceAccessor(ctx, parent, args.Name)
```

Comment:
General question for @sipsma - do we still need this "accessor" property? Now we've got the secret separation via the `SecretsStore` is this still something that we need?

Or is it still required to ensure we still get actually unique values per module?